server/ai_http.go

```diff
@@ -139,8 +155,85 @@ func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 			}
 		}()
 
-		// TODO subscribe to the subscribeUrl for output monitoring
+		// TODO: Use params temporarily to pass the host from gateway
+		host := req.PublishUrl
+		pub, err := common.AppendHostname(pubUrl, host)
+		if err != nil {
+			respondWithError(w, fmt.Errorf("invalid publish URL: %w", err).Error(), http.StatusBadRequest)
+			return
+		}
+		sub, err := common.AppendHostname(subUrl, host)
+		if err != nil {
+			respondWithError(w, fmt.Errorf("invalid subscribe URL: %w", err).Error(), http.StatusBadRequest)
+			return
+		}
+		control, err := common.AppendHostname(controlUrl+"-control", host)
+		if err != nil {
+			respondWithError(w, fmt.Errorf("invalid control URL: %w", err).Error(), http.StatusBadRequest)
+			return
+		}
+
+		// Prepare request to worker
+		workerReq := worker.LiveVideoToVideoParams{
+			ModelId:      req.ModelId,
+			PublishUrl:   sub, // SubscribeUrl is the publish url for the worker
+			SubscribeUrl: pub, // PublishUrl is the subscribe url for the worker
+			ControlUrl:   &control,
+			Params:       req.Params,
+		}
+
+		// append params to the request
+		queryParams := r.FormValue("query")
+		qp, err := url.ParseQuery(queryParams)
+		if err != nil {
+			respondWithError(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		// copy the params if they exist
+		var paramsMap map[string]interface{}
+		params := qp.Get("params")
+		if params != "" {
+			if err := json.Unmarshal([]byte(params), &paramsMap); err != nil {
+				respondWithError(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+			workerReq.Params = &paramsMap
+		}
+
+		// Send request to the worker
+		_, err = orch.LiveVideoToVideo(ctx, requestID, workerReq)
+		if err != nil {
+			if monitor.Enabled {
+				monitor.AIProcessingError(err.Error(), pipeline, modelID, ethcommon.Address{}.String())
+			}
+
+			pubCh.Close()
+			subCh.Close()
+			controlPubCh.Close()
+
+			respondWithError(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		// Prepare the response
+		resp := &worker.LiveVideoToVideoResponse{
+			PublishUrl:   workerReq.PublishUrl,
+			SubscribeUrl: workerReq.SubscribeUrl,
+		}
+
+		// TODO: Fix the nullable type in the worker response
+		if workerReq.ControlUrl != nil {
```

Comment:
What is this? Why ControlUrl is not passed from the worker the same as in the case of `PublishUrl`?

Reply:
ControlUrl was nullable on the ai-runner, I have solved this in https://github.com/livepeer/ai-worker/pull/286

Reply:
We've reverted this back to an optional field